From c6ddb3e4ca72f6ec8662f8a18674eb4d861561b8 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon, 10 Feb 2014 13:03:50 +0000
Subject: [PATCH 3/4] always define _GNU_SOURCE for linux

We are using various extenstions that the spec say depends on _GNU_SOURCE,
for example unshare, CPU_SET, CPU_ZERO, cpu_set_t. We enable those always
for linux and we never unset it.

Signed-off-by: Natanael Copa <ncopa@alpinelinux.org>
---
 uwsgi.h | 18 +++++-------------
 1 file changed, 5 insertions(+), 13 deletions(-)

diff --git a/uwsgi.h b/uwsgi.h
index b3ce4f7..3131a0f 100644
--- a/uwsgi.h
+++ b/uwsgi.h
@@ -149,29 +149,22 @@ extern "C" {
 #endif
 #endif
 
+#ifdef __linux__
 #ifndef _GNU_SOURCE
 #define _GNU_SOURCE
 #endif
-#include <stdio.h>
-#ifdef __UCLIBC__
-#include <sched.h>
+#ifndef __USE_GNU
+#define __USE_GNU
+#endif
 #endif
-#undef _GNU_SOURCE
 
+#include <stdio.h>
 #include <stdlib.h>
 #include <stddef.h>
 #include <signal.h>
 #include <math.h>
 
 #include <sys/types.h>
-#ifdef __linux__
-#ifndef _GNU_SOURCE
-#define _GNU_SOURCE
-#endif
-#ifndef __USE_GNU
-#define __USE_GNU
-#endif
-#endif
 #include <sys/socket.h>
 #include <net/if.h>
 #ifdef __linux__
@@ -179,7 +172,6 @@ extern "C" {
 #define MSG_FASTOPEN   0x20000000
 #endif
 #endif
-#undef _GNU_SOURCE
 #include <netinet/in.h>
 
 #include <termios.h>
-- 
1.8.5.3

